pkg/kp/rcstore/consul_store.go

```diff
@@ -26,39 +24,66 @@ type consulKV interface {
 	Get(key string, opts *api.QueryOptions) (*api.KVPair, *api.QueryMeta, error)
 	Keys(prefix string, separator string, opts *api.QueryOptions) ([]string, *api.QueryMeta, error)
 	List(prefix string, opts *api.QueryOptions) (api.KVPairs, *api.QueryMeta, error)
-	Put(pair *api.KVPair, opts *api.WriteOptions) (*api.WriteMeta, error)
+	CAS(pair *api.KVPair, opts *api.WriteOptions) (bool, *api.WriteMeta, error)
+	DeleteCAS(pair *api.KVPair, opts *api.WriteOptions) (bool, *api.WriteMeta, error)
 	Acquire(pair *api.KVPair, opts *api.WriteOptions) (bool, *api.WriteMeta, error)
-	DeleteTree(prefix string, opts *api.WriteOptions) (*api.WriteMeta, error)
 }
 
 type consulStore struct {
 	applicator labels.Applicator
 	kv         consulKV
 	logger     logging.Logger
+	retries    int
+}
+
+// TODO: combine with similar CASError type in pkg/labels
+type CASError string
+
+func (e CASError) Error() string {
+	return fmt.Sprintf("Could not compare-and-set key %q", string(e))
```

Comment:
super nit i'm used to this being called "check-and-set" and seems like that's the terminology consul's documentation uses